pkg/helm/client/client.go

```diff
@@ -98,16 +101,24 @@ func NewRESTClientGetter(mgr manager.Manager, ns string) (genericclioptions.REST
 
 var _ kube.Interface = &ownerRefInjectingClient{}
 
-func NewOwnerRefInjectingClient(base kube.Client, ownerRef metav1.OwnerReference) kube.Interface {
-	return &ownerRefInjectingClient{
-		refs:   []metav1.OwnerReference{ownerRef},
-		Client: base,
+func NewOwnerRefInjectingClient(base kube.Client, restMapper meta.RESTMapper,
+	cr *unstructured.Unstructured) (kube.Interface, error) {
+
+	if cr.GetObjectKind().GroupVersionKind().Empty() || cr.GetName() == "" || cr.GetUID() == "" {
```

Comment:
also need to to a `nil` check on `cr` and `cr.GetObjectKind()`. Otherwise, this could panic.